src/main/impl/Config.cpp

```diff
@@ -0,0 +1,422 @@
+#include <boost/algorithm/string.hpp>
+
+#include <main/Config.h>
+
+#include <fstream>
+#include <iostream>
+
+boost::json::object
+parseConfig(std::string const& filename)
+{
+    try
+    {
+        std::ifstream in(filename, std::ios::in | std::ios::binary);
+        if (in)
+        {
+            std::stringstream contents;
+            contents << in.rdbuf();
+            in.close();
+            std::cout << contents.str() << std::endl;
+            boost::json::value value = boost::json::parse(contents.str());
+            return value.as_object();
+        }
+    }
+    catch (std::exception const& e)
+    {
+        std::cout << e.what() << std::endl;
+    }
+
+    throw std::runtime_error(
+        "Could not parse JSON configuration. Verify it is valid JSON format");
+}
+
+struct cfgOPTIONAL;
+struct cfgREQUIRED;
+
+template <typename T, typename F>
+void
+add(T& value, boost::json::value& json, F&& parser)
+{
+    try
+    {
+        value = parser(json);
+    }
+    catch (std::runtime_error const& e)
+    {
+        throw std::runtime_error(
+            "Could not parse field " + boost::json::serialize(json) + "\n" +
+            e.what());
+    }
+}
+
+template <typename T, typename F, typename Def>
+void
+add(T& value, boost::json::value& json, F&& parser, Def def)
```

Comment:
I'd change `Def` to `V` and `def` to `Default`.

Reply:
`default` is a keyword in cpp

will use `dfault`.